### PR TITLE
Select or Capture new profile image, Setup Axios

### PR DESCRIPTION
**Pull request by pointmekin**
# Notes
- Updated the profile edit page types to match the new user
- Setup Axios
- Setup `useAuthenticationHook` to set and get JSON web token.
- Request android permission to access the camera (can't test on iOS yet due to the emulator having no camera)
- Select an image or take a new photo functionality in the profile edit page

https://user-images.githubusercontent.com/53035529/160098065-dbbf6ea4-b10b-4be9-8911-65043c5ec2b6.mov

### CODE REVIEW

File: src/screens/ProfileEditScreen.tsx [hunk 1]
```diff
@@ -82,18 +90,50 @@ const ProfileEditScreen = () => {
     return momentDate.format('MM/DD/YYYY');
   };
 
+  const takePicture = async () => {
+    const result: ImagePickerResponse = await launchCamera(CameraPhotoOptions);
+    setNewProfileImage(result);
+    console.log(result);
```
Comment (chuwong35122) [inline src/screens/ProfileEditScreen.tsx]: remove console.log

File: src/utils/permission.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  mediaType: 'photo'
+};
+
+export const requestCameraPermission = async () => {
+  if (Platform.OS === 'android') {
+    try {
+      const granted = await PermissionsAndroid.request(
+        PermissionsAndroid.PERMISSIONS.CAMERA,
+        {
+          title: 'Cool Photo App Camera Permission',
```
Comment (chuwong35122) [inline src/utils/permission.ts]: maybe change title, description...

File: src/hooks/useAuthentication.tsx [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import AsyncStorage from '@react-native-async-storage/async-storage';
+export const useAuthentication = () => {
+  const setToken = async (token: string) => {
+    AsyncStorage.setItem('token', token);
+  };
+  const getToken = async () => {
+    return AsyncStorage.getItem('token') ?? '';
+  };
+  return { setToken, getToken };
+};
```
Comment (chuwong35122) [inline src/hooks/useAuthentication.tsx:10]: I've created token state in UserContext, can you call this hook there also

**Review by chuwong35122 — CHANGES_REQUESTED** [review]

**Review by bweam — APPROVED** [review]

**Review by chuwong35122 — APPROVED** [review]
Should we show native-base's toast component when an error is watched?

**Pull request merged**